client/post-editor/editor-html-toolbar/index.jsx

```diff
@@ -0,0 +1,309 @@
+		this.setState( { openTags: openTags.concat( tag.name ) } );
+	}
+
+	insertHtmlTagClose( tag ) {
+		const { openTags } = this.state;
+		const { before, after } = this.splitEditorContent();
+		this.updateEditorContent( before + this.closeHtmlTag( tag ) + after );
+		this.setState( { openTags: openTags.filter( openTag => openTag !== tag.name ) } );
+	}
+
+	insertHtmlTagOpenClose( tag ) {
+		const { before, inner, after } = this.splitEditorContent();
+		this.updateEditorContent( before + this.openHtmlTag( tag ) + inner + this.closeHtmlTag( tag ) + after );
+	}
+
+	insertHtmlTagSelfClosed( tag ) {
+		const { before, inner, after } = this.splitEditorContent();
+		const selfClosedTag = `<${ tag.name }${ this.attributesToString( tag.attributes ) } />`;
+		const content = tag.options && tag.options.newLine ? '\n' + selfClosedTag + '\n\n' : selfClosedTag;
+		this.updateEditorContent( before + inner + content + after );
+	}
+
+	insertHtmlTagWithText( tag ) {
+		const { before, after } = this.splitEditorContent();
+		this.updateEditorContent( before + this.openHtmlTag( tag ) + tag.options.text + this.closeHtmlTag( tag ) + after );
+	}
+
+	insertCustomContent( content ) {
+		const { before, inner, after } = this.splitEditorContent();
+		this.updateEditorContent( before + inner + content + after );
+	}
+
+	insertHtmlTag( tag ) {
+		const { content: {
+			selectionEnd,
+			selectionStart,
+		} } = this.props;
+		const { openTags } = this.state;
+		if ( selectionEnd === selectionStart ) {
+			const isTagOpen = -1 !== openTags.indexOf( tag.name );
+			isTagOpen ? this.insertHtmlTagClose( tag ) : this.insertHtmlTagOpen( tag );
+		} else {
+			this.insertHtmlTagOpenClose( tag );
+		}
+	}
+
+	onClickBold = () => {
+		this.insertHtmlTag( { name: 'strong' } );
+	}
+
+	onClickItalic = () => {
+		this.insertHtmlTag( { name: 'em' } );
+	}
+
+	onClickLink = ( attributes, text ) => {
+		if ( text ) {
+			this.insertHtmlTagWithText( { name: 'a', attributes, options: { text } } );
+		} else {
+			this.insertHtmlTagOpenClose( { name: 'a', attributes } );
+			// Move the cursor inside <a></a>
+			this.setCursorPosition( this.props.content.selectionEnd, -4 );
+		}
+	};
+
+	onClickQuote = () => {
+		this.insertHtmlTag( { name: 'blockquote' } );
+	}
+
+	onClickDelete = () => {
+		const datetime = this.props.moment().format();
+		this.insertHtmlTag( { name: 'del', attributes: { datetime } } );
+	}
+
+	onClickInsert = () => {
+		const datetime = this.props.moment().format();
+		this.insertHtmlTag( { name: 'ins', attributes: { datetime } } );
+	}
+
+	onClickImage = attributes => {
+		this.insertHtmlTagSelfClosed( { name: 'img', attributes } );
+	}
+
+	onClickUnorderedList = () => {
+		this.insertHtmlTag( { name: 'ul', options: { newLine: true } } );
+	}
+
+	onClickOrderedList = () => {
+		this.insertHtmlTag( { name: 'ol', options: { newLine: true } } );
+	}
+
+	onClickListItem = () => {
+		this.insertHtmlTag( { name: 'li' } );
```

Comment:
I was sure I did that! Likely got lost in one of the refactor I did on the code there.